lib/streamlit/elements/widgets/button_group.py

```diff
@@ -403,61 +403,55 @@ def _pills(
-            it parsed to the frontend."""
-            transformed = format_func(option) if format_func else str(option)
-            transformed_parts = transformed.split(" ")
-            icon: str | None = None
-            if len(transformed_parts) > 0:
-                maybe_icon = transformed_parts[0].strip()
-                try:
-                    # we only want to extract material icons because we treat them
-                    # differently than emojis visually
-                    if maybe_icon.startswith(":material"):
-                        icon = validate_material_icon(maybe_icon)
-                        # reassamble the option string without the icon - also
-                        # works if len(transformed_parts) == 1
-                        transformed = " ".join(transformed_parts[1:])
-                except StreamlitAPIException:
-                    # we don't have a valid icon or emoji, so we just pass
-                    pass
-            return ButtonGroupProto.Option(
-                content=transformed,
-                content_icon=icon,
-            )
-
-        indexable_options = convert_to_sequence_and_check_comparable(options)
-        default_values = get_default_indices(indexable_options, default)
-
-        serde: SingleOrMultiSelectSerde[V] = SingleOrMultiSelectSerde[V](
-            indexable_options, default_values, selection_mode
-        )
-        res = self._button_group(
-            indexable_options,
-            key=key,
-            default=default_values,
+    ):
+        return self._internal_button_group(
+            options,
+            label=label,
             selection_mode=selection_mode,
-            disabled=disabled,
-            format_func=_transformed_format_func,
-            serializer=serde.serialize,
-            deserializer=serde.deserialize,
+            default=default,
+            format_func=format_func,
+            key=key,
+            help=help,
+            style="pills",
             on_change=on_change,
             args=args,
             kwargs=kwargs,
-            style="pills",
-            label=label,
+            disabled=disabled,
             label_visibility=label_visibility,
-            help=help,
         )
 
-        if selection_mode == "multiple":
-            return res.value
-
-        return res.value
+    @gather_metrics("segmented_control")
+    def segmented_control(
+        self,
+        label: str,
+        options: OptionSequence[V],
+        *,
+        selection_mode: Literal["single", "multiple"] = "single",
+        default: Sequence[V] | V | None = None,
+        format_func: Callable[[Any], str] | None = None,
+        key: str | int | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        disabled: bool = False,
+        label_visibility: LabelVisibility = "visible",
+    ):
```

Comment:
it could make sense to add a first version of a docstring (likely you can just copy & paste the one from pills) so that we have at least something and then @sfc-gh-dmatthews could chime in and make it great 🙂